streams/src/main/java/org/apache/kafka/streams/kstream/TimeWindows.java

```diff
@@ -159,11 +164,16 @@ public TimeWindows advanceBy(final long advanceMs) {
      *
      * @param advance The advance interval ("hop") of the window, with the requirement that {@code 0 < advance.toMillis() <= sizeMs}.
      * @return a new window definition with default maintain duration of 1 day
-     * @throws IllegalArgumentException if the advance interval is negative, zero, or larger-or-equal the window size
+     * @throws IllegalArgumentException if the advance interval is negative, zero, or larger than the window size
      */
     @SuppressWarnings("deprecation") // will be fixed when we remove segments from Windows
     public TimeWindows advanceBy(final Duration advance) {
-        ApiUtils.validateMillisecondDuration(advance, "advance");
+        try {
+            ApiUtils.validateMillisecondDuration(advance, "advance");
+        } catch (final IllegalArgumentException e) {
+            throw new IllegalArgumentException(String.format("Advance interval(%d ms) must lie " +
+                    "within window duration (0, %d].", advanceMs, sizeMs), e);
```

Comment:
`advanceMs` is not the same as provided input parameter `advance` -- this would make the error message miss leading.